homeassistant/components/hue/__init__.py

```diff
@@ -53,14 +54,47 @@
 
 
 async def async_setup(hass, config):
-    """Set up the Hue platform."""
+
+    async def hue_activate_scene(call, skip_reload=True):
+
+        # Get parameters
+        group_name = call.data[ATTR_GROUP_NAME]
+        scene_name = call.data[ATTR_SCENE_NAME]
+
+        # Call the set scene function on each bridge
+        results = []
+        for dummy_host, bridge in hass.data[DOMAIN].items():
+            result = await bridge.hue_activate_scene(call,
```

Comment:
Collect all coroutines in a list and then await that list of coroutines, so it happens in parallel.

```python
tasks = [bridge.hue_activate_scene(…) for bridge in hass.data[DOMAIN].values()]

results = await asyncio.gather(tasks)

if not any(result is False for result in results):
    return

if skip_reload:
    …
```

Reply:
Great suggestion - thanks. Still getting used to asyncio ;p